src/e2e/puppeteer/helpers/setSelection.ts

```diff
@@ -0,0 +1,22 @@
+import { page } from '../setup'
+
+/** Sets the selection of the text in the editable. */
+const setSelection = async (start: number, end: number) => {
+  await page.evaluate(
+    (start: number, end: number) => {
+      const selection = window.getSelection()
+      const range = document.createRange()
+      const textNode = document.querySelector('[contenteditable="true"]')?.firstChild
+      if (textNode) {
```

Comment:
If there is no `textNode`, this function will return silently. This is a bad practice in tests. In tests, we want to fail hard when values are not what they are expected to be. Therefore, you should throw an exception if there is no text node.